src/replication.c

```diff
@@ -839,11 +1008,36 @@ zvol_handshake(spec_t *spec, replica_t *replica)
 		REPLICA_ERRLOG("make_socket_non_blocking() failed errno:%d\n", errno);
 		exit(EXIT_FAILURE);
 	}
-	update_replica_entry(spec, replica, iofd, mgmt_ack_data->ip, mgmt_ack_data->port);
+	update_replica_entry(spec, replica, iofd, repl_info->ip, repl_info->port);
 
 	return 0;
 }
 
+/*
+ * queue_handshake_read_msg add mgmt_cmd related to HANDSHAKE
+ * (which is already sent to replica) in mgmt_cmd_queue of replica
+ * for upcoming read
+ */
+static void
+queue_handshake_read_msg(replica_t *replica)
```

Comment:
why don't we queue for sending handshake msg?

Reply:
there was a separate function send_mgmtio for sending handshake message so I have avoided queueing handshake msg. We can surely queue handshake message in mgmt_cmd_queue. I will update it accordingly.